src/libNode/FinalBlockProcessing.cpp

```diff
@@ -1518,7 +1572,7 @@ bool Node::ProcessFinalBlock(const vector<unsigned char>& message,
     unsigned int cur_offset = offset;
 
     // Initialize it with 255
-    uint8_t shard_id = (uint8_t)-1;
+    uint32_t shard_id = (uint32_t)-1;
```

Comment:
I did some search over the codebase and it turns out that this the only place we assign a maximum value like this. Can we have a try of [numeric_limits](http://en.cppreference.com/w/cpp/types/numeric_limits) and replace the use of this?

Reply:
Noted